File: web-app/src/utils/controller/userController.js
import { api } from "src/boot/axios";

let searchUsers = (search) => {
  const param = new FormData();
  param.append("search", search);
  return api.post("user/search-user", param);
};

let createUser = (user) => {
  return api.post("/user", user);
};

let getUserById = (userId) => {
  const data = api.get(`/user/${userId}`);
  return data;
};

let updateUser = (userId, user) => {
  return api.put(`/user/${userId}`, user);
};

let deleteUser = (userId) => {
  return api.delete(`/user/${userId}`);
};

export const userController = {
  searchUsers,
  createUser,
  getUserById,
  updateUser,
  deleteUser
};

File: bank-account-project/src/main/java/com/example/bank_account_project/controller/UserController.java
package com.example.bank_account_project.controller;

import com.example.bank_account_project.dto.UserDto;
import com.example.bank_account_project.service.UserService;
import com.example.bank_account_project.utils.validable.Validable;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/user")
@RequiredArgsConstructor
public class UserController {

    private final UserService usersService;

    @PostMapping("/search-user")
    public ResponseEntity<List<UserDto>> searchUsers(@RequestParam("search") String search) {
        List<UserDto> users = usersService.searchUsers(search);
        return ResponseEntity.ok(users);
    }

    @PostMapping
    public ResponseEntity<Validable<UserDto>> createUser(@RequestBody UserDto usersDto) {
        Validable<UserDto> createdUser = usersService.createUser(usersDto);
        return ResponseEntity.ok(createdUser);
    }

    @GetMapping("/{id}")
    public ResponseEntity<UserDto> getUserById(@PathVariable Long id) {
        return usersService.getUserById(id)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    @PutMapping("/{id}")
    public ResponseEntity<Validable<UserDto>> updateUser(@PathVariable Long id, @RequestBody UserDto usersDto) {
        Validable<UserDto> updatedUser = usersService.updateUser(id, usersDto);
        return ResponseEntity.ok(updatedUser);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteUser(@PathVariable Long id) {
        usersService.deleteUser(id);
        return ResponseEntity.noContent().build();
    }

}
